src/dataExplorer/components/FluxQueryBuilder.tsx

```diff
@@ -1,38 +1,105 @@
+import LeftArrow from 'assets/images/collapse-left.svg'
+import RightArrow from 'assets/images/collapse-right.svg'
 
-const INITIAL_VERT_RESIZER_HANDLE = 0.2
+const INITIAL_LEFT_VERT_RESIZER_HANDLE = 0.2
+const INITIAL_RIGHT_VERT_RESIZER_HANDLE = 0.8
 
 const FluxQueryBuilder: FC = () => {
   const [vertDragPosition, setVertDragPosition] = useState([
-    INITIAL_VERT_RESIZER_HANDLE,
+    INITIAL_LEFT_VERT_RESIZER_HANDLE,
+    INITIAL_RIGHT_VERT_RESIZER_HANDLE,
   ])
+  const [showLeft, toggleLeftPanel] = useState(true)
+  const [showRight, toggleRightPanel] = useState(true)
+
+  const adjustMiddlePanel = grow => {
+    if (!grow) {
+      return
+    }
+    const el = document.querySelector(
+      '[data-testid="flux-query-builder-middle-panel"]'
+    )
+    if (el) {
+      (el as any).style.flexGrow = 0.8
+    }
+  }
 
   return (
     <QueryProvider>
-      <DraggableResizer
-        handleOrientation={Orientation.Vertical}
-        handlePositions={vertDragPosition}
-        onChangePositions={setVertDragPosition}
-      >
-        <DraggableResizer.Panel>
-          <Schema />
-        </DraggableResizer.Panel>
-        <DraggableResizer.Panel className="new-data-explorer-rightside">
-          <ResultsProvider>
-            <ResultsPane />
-          </ResultsProvider>
-        </DraggableResizer.Panel>
-      </DraggableResizer>
+      <EditorProvider>
+        <DraggableResizer
+          handleOrientation={Orientation.Vertical}
+          handlePositions={vertDragPosition}
+          onChangePositions={setVertDragPosition}
+        >
+          <DraggableResizer.Panel style={showLeft ? {} : {display: 'none'}}>
+            <Schema />
+          </DraggableResizer.Panel>
+          <DraggableResizer.Panel
+            testID="flux-query-builder-middle-panel"
+            className="new-data-explorer-rightside"
+          >
+            <FlexBox
+              direction={FlexDirection.Row}
+              alignItems={AlignItems.FlexStart}
+              style={{height: '100%', width: '100%', position: 'absolute'}}
+            >
+              <FlexBox.Child>
+                <span
+                  onClick={() => {
+                    toggleLeftPanel(!showLeft)
+                    adjustMiddlePanel(showLeft)
+                  }}
+                  style={{marginLeft: '-20px', float: 'left'}}
+                >
+                  <img src={showLeft ? LeftArrow : RightArrow} />
+                </span>
+              </FlexBox.Child>
+              <ResultsProvider>
+                <ResultsPane />
+              </ResultsProvider>
+              <FlexBox.Child>
+                <span
+                  onClick={() => {
+                    toggleRightPanel(!showRight)
+                    adjustMiddlePanel(showRight)
+                  }}
+                  style={{
+                    marginRight: '-20px',
+                    float: 'right',
+                    zIndex: '999',
+                    position: 'absolute',
```

Comment:
For the right side's collapsing icon, I had to set the zIndex to make it actionable above the panelDraggableHandle.

Reply:
Removed the collapsing panel. Resolved.